src/Utils.hs

```diff
@@ -81,10 +81,7 @@ parsecReader lbl p = OA.eitherReader (\s -> first (prerr s) . parse (p <* (eof <
 
 -- | lift binary combining operation to 'Maybe's
 onMaybes :: (a -> a -> a) -> Maybe a -> Maybe a -> Maybe a
```

Comment:
Generalises to any `Alternative`, though here a more constrained type probably makes more sense.